### PR TITLE
Move react rules out of itwinjs-recommended

### PR DESCRIPTION
**Pull request by paulius-valiunas**

### CODE REVIEW

**Review by pmconne — APPROVED** [review]
Did it have any noticeable impact on lint time for e.g. core-frontend?

**Review by NancyMcCallB — APPROVED** [review]

**Comment by paulius-valiunas** [comment]
@NancyMcCallB please take a look at the lint errors in ui-test-app - I'd be fine with disabling those rules for the entire test app if you feel like they're not helpful.

File: extensions/map-layers/package.json [hunk 1]
```diff
@@ -111,7 +111,7 @@
     "plugins": [
       "@itwin"
     ],
-    "extends": "plugin:@itwin/itwinjs-recommended"
+    "extends": "plugin:@itwin/ui"
```
Comment (calebmshafer) [inline extensions/map-layers/package.json:114]: Can you put this in the next version.md for people that use our configs to know what they need to update? I expect come of the pkgs in viewer components react will need to update as well.

**Review by calebmshafer — APPROVED** [review]
LGTM other than the comment about adding the docs.

File: extensions/map-layers/package.json (hunk 1 above)
Comment (paulius-valiunas) [reply extensions/map-layers/package.json:114]: We've always had these two separate configs and if anyone used itwinjs-recommended for a UI package, that was already a mistake. Do you still want me to explicitly point it out what each config is for?

Comment (calebmshafer) [reply extensions/map-layers/package.json:114]: ah okay, nope you're good then. resolving this comment.

**Comment by paulius-valiunas** [comment]
@pmconne I just compared the timings with a previous run. There isn't a massive improvement, but these 3 packages had a >20% speed up:

@itwin/core-geometry
@itwin/core-common
@itwin/core-frontend

Only @itwin/express-server and @itwin/appui-test-providers slowed down more than 10%, I'm not sure what's up with express-server. Adding all the packages together, there was an overall speed up of 5%. I'd say everything is within tolerance, the 3 packages with >20% speed up are probably the only ones that could potentially be noticeable. However, it's a very small difference compared to all the other packages, and it's further reduced by parallel execution

**Comment by pmconne** [comment]
> these 3 packages had a >20% speed up.

That's great. core-frontend in particular got much slower after we switched to ESLint.

**Comment by pmconne** [comment]
@MarcBedard8 @mdastous-bentley please review.

**Review by mdastous-bentley — APPROVED** [review]

**Comment by calebmshafer** [comment]
@NancyMcCallB could you take a look at the failures in ui-test-app?

**Comment by NancyMcCallB** [comment]
> @NancyMcCallB could you take a look at the failures in ui-test-app?

Sorry! lost this in my github mail. Looking at it now!

**Pull request merged**